event/events.go

```diff
@@ -118,5 +135,25 @@ func GenerateSubId() (string, error) {
 	}
 	rStr := hex.EncodeToString(b)
 	return strings.ToUpper(rStr), nil
+}
 
+func mapToOurEventData(eventData anyEventData) (txs.EventData, error) {
+	// While we depend on go-events in the way we do, we don't have much choice
```

Comment:
I'm really not sure there is a nicer way to do this. In essence `anyEventData` is wrapping around go-events `EventData`. I could embed `go_events.EventData` in `anyEventData`, but bear in mind that is just `interface{}` so the effect would be the same.

If we have decided that we want to have these event types in our code (I think we have) then this allows us to do more complex mapping/normalisation of events if we ever need to, for example for other consensus engines. But using a type switch like this is regrettable how this sort of thing is done in go, I think.

I'm open to suggestions if you know of a better pattern or have something specific I haven't thought of when you say 'wraps around go-events'